Classes/Auth/TMOAuth.m

```diff
@@ -75,6 +90,37 @@ - (id)initWithURL:(NSURL *)URL method:(NSString *)method postParameters:(NSDicti
 
 #pragma mark - Private
 
+NSMutableDictionary *generateOAuthParameters(TMOAuth *tmOAuth,
```

Comment:
The fact that this is a C function feels a little weird because it's a little tough to context switch from Objective-C to C midway through this file. What about making this a static Objective-C method instead? Also we might consider changing the name from the more procedural "generateOAuthParameters" to the more object-oriented (and Objective-C idiomatic) name `OAuthParametersFrom...`:

```objc
+ (NSMutableDictionary *)OAuthParametersFromURL:(NSURL *)URL
                                         method:(NSString *)method
                                 postParameters:(NSDictionary *)postParameters
                                          nonce:(NSString *)nonce
                                    consumerKey:(NSString *)consumerKey
                                 consumerSecret:(NSString *)consumerSecret
                                          token:(NSString *)token
                                    tokenSecret:(NSString *)tokenSecret
                                      timestamp:(NSString *)timestamp {
  /* ... */
}`